### ISSUE TITLE
Issue with los  script

### ISSUE THREAD

**Issue by clementinegrethen** [wzzheng/StreamVGGT#23]
Hello,
Thank you for the repo !
I have seen few erros in the loss script while I have tried to fine tune the VGGT model, You have forgotten to import some function like check_and_fix_inf_nan ( I took the one from vggt official repo) and also "from streamvggt.utils.pose_enc import extri_intri_to_pose_encoding
". For my dataset I got some errors when using the dataloader based on multiviewbase espacially for the format of camera pose, I have added :

# THE ONE FROM VGGT REPO
def check_and_fix_inf_nan(input_tensor, loss_name="default", hard_max=100):
    """
    Checks if 'input_tensor' contains inf or nan values and clamps extreme values.
    
    Args:
        input_tensor (torch.Tensor): The loss tensor to check and fix.
        loss_name (str): Name of the loss (for diagnostic prints).
        hard_max (float, optional): Maximum absolute value allowed. Values outside 
                                  [-hard_max, hard_max] will be clamped. If None, 
                                  no clamping is performed. Defaults to 100.
    """
    if input_tensor is None:
        return input_tensor
    
    # Check for inf/nan values
    has_inf_nan = torch.isnan(input_tensor).any() or torch.isinf(input_tensor).any()
    if has_inf_nan:
        logging.warning(f"Tensor {loss_name} contains inf or nan values. Replacing with zeros.")
        input_tensor = torch.where(
            torch.isnan(input_tensor) | torch.isinf(input_tensor),
            torch.zeros_like(input_tensor),
            input_tensor
        )

    # Apply hard clamping if specified
    if hard_max is not None:
        input_tensor = torch.clamp(input_tensor, min=-hard_max, max=hard_max)

    return input_tensor
# My own functions 
def _rot6d_to_rotmat(r6: torch.Tensor) -> torch.Tensor:
    # r6: (..., 6)
    a1 = F.normalize(r6[..., 0:3], dim=-1)
    a2 = r6[..., 3:6]
    b2 = F.normalize(a2 - (a1 * a2).sum(dim=-1, keepdim=True) * a1, dim=-1)
    b3 = torch.cross(a1, b2, dim=-1)
    return torch.stack((a1, b2, b3), dim=-2)  # (..., 3, 3)

def _rotmat_to_quat_wxyz(R: torch.Tensor) -> torch.Tensor:
    # R: (..., 3, 3) → quat (w,x,y,z)
    t = R[..., 0, 0] + R[..., 1, 1] + R[..., 2, 2]
    qw = torch.sqrt(torch.clamp(1.0 + t, min=1e-8)) * 0.5
    qx = (R[..., 2, 1] - R[..., 1, 2]) / (4.0 * qw + 1e-8)
    qy = (R[..., 0, 2] - R[..., 2, 0]) / (4.0 * qw + 1e-8)
    qz = (R[..., 1, 0] - R[..., 0, 1]) / (4.0 * qw + 1e-8)
    q = torch.stack((qw, qx, qy, qz), dim=-1)
    return F.normalize(q, dim=-1)

and modified 
    def compute_pose_loss(self, gt_poses, pred_poses, masks=None):
        # gt: liste de (Bx3, Bx4) ; pred: liste de (Bx3, Bx{4|6|9})
        gt_trans = torch.stack([gt[0] for gt in gt_poses], dim=1)   # BxNx3
        gt_quats = torch.stack([gt[1] for gt in gt_poses], dim=1)   # BxNx4
        pred_trans = torch.stack([pr[0] for pr in pred_poses], dim=1)  # BxNx3
        pred_rot   = torch.stack([pr[1] for pr in pred_poses], dim=1)  # BxNx(4|6|9)
    
        # --- conversion si rot6d (6) ou rotmat aplati (9)
        if pred_rot.shape[-1] in (6, 9):
            if pred_rot.shape[-1] == 6:
                R = _rot6d_to_rotmat(pred_rot)                      # BxNx3x3
            else:  # 9
                R = pred_rot.view(*pred_rot.shape[:-1], 3, 3)
            pred_quats = _rotmat_to_quat_wxyz(R)                    # BxNx4
        else:
            pred_quats = pred_rot                                    # déjà quat
    
        if masks is None:
            pose_loss = (torch.norm(pred_trans - gt_trans, dim=-1).mean()
                         + torch.norm(pred_quats - gt_quats, dim=-1).mean())
        else:
            if not any(masks):
                return torch.tensor(0.0, device=pred_trans.device)
            pose_loss = (torch.norm(pred_trans - gt_trans, dim=-1)[masks].mean()
                         + torch.norm(pred_quats - gt_quats, dim=-1)[masks].mean())
        return pose_loss
Feel free to tell me  if you have some recommandations !
